test/integration/controlplane/transformation/transformation_test.go

```diff
@@ -401,3 +412,51 @@ func (e *transformTest) printMetrics() error {
 
 	return nil
 }
+
+func mustBeHealthy(t kubeapiservertesting.Logger, checkName, wantBodyContains string, clientConfig *rest.Config, excludes ...string) {
+	t.Helper()
+	var restErr error
+	pollErr := wait.PollImmediate(2*time.Second, wait.ForeverTestTimeout, func() (bool, error) {
+		body, ok, err := getHealthz(checkName, clientConfig, excludes...)
+		restErr = err
+		if err != nil {
+			return false, err
+		}
+		return ok && strings.Contains(body, wantBodyContains), nil
+	})
+
+	if pollErr == wait.ErrWaitTimeout {
+		t.Fatalf("failed to get the expected healthz status of OK for check: %s, error: %v, debug inner error: %v", checkName, pollErr, restErr)
+	}
+}
+
+func mustBeUnHealthy(t kubeapiservertesting.Logger, checkName, wantBodyContains string, clientConfig *rest.Config, excludes ...string) {
+	t.Helper()
+	var restErr error
+	pollErr := wait.PollImmediate(2*time.Second, wait.ForeverTestTimeout, func() (bool, error) {
+		body, ok, err := getHealthz(checkName, clientConfig, excludes...)
+		restErr = err
+		if err != nil {
+			return false, err
+		}
+		return !ok && strings.Contains(body, wantBodyContains), nil
+	})
+
+	if pollErr == wait.ErrWaitTimeout {
+		t.Fatalf("failed to get the expected healthz status of !OK for check: %s, error: %v, debug inner error: %v", checkName, pollErr, restErr)
+	}
+}
+
+func getHealthz(checkName string, clientConfig *rest.Config, excludes ...string) (string, bool, error) {
+	client, err := kubernetes.NewForConfig(clientConfig)
+	if err != nil {
+		return "", false, fmt.Errorf("failed to create a client: %v", err)
+	}
+
+	req := client.CoreV1().RESTClient().Get().AbsPath(fmt.Sprintf("/healthz%v", checkName)).Param("verbose", "true")
+	for _, exclude := range excludes {
+		req.Param("exclude", exclude)
+	}
+	body, err := req.DoRaw(context.TODO()) // we can still have a response body during an error case
+	return string(body), err == nil, nil
```

Comment:
is `err == nil` the right check for health, or 200 status code?

Reply:
`DoRaw` does not give access to the `Result` to check the status code from what I can tell, and I needed that function to avoid the rest client trying to decode the unstructured body.  `DoRaw` does check for a status `2xx` and will otherwise return an error:

https://github.com/kubernetes/kubernetes/blob/548d135ea6b3b9a68d31c981b65e1f9e9643ea58/staging/src/k8s.io/client-go/rest/request.go#L1017-L1025

Reply:
oh, ok, I got mixed up with http.Client which only returns errors for network communication issues, not non-2xx responses